File: code/frontend/src/app/shared/service/admin/commun/SecteurAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {SecteurDto} from 'src/app/shared/model/commun/Secteur.model';
import {SecteurCriteria} from 'src/app/shared/criteria/commun/SecteurCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class SecteurAdminService extends AbstractService<SecteurDto, SecteurCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): SecteurDto {
        return new SecteurDto();
    }

    public constrcutCriteria(): SecteurCriteria {
        return new SecteurCriteria();
    }

    get API() {
        return environment.apiUrlUnivservice + 'admin/secteur/';
    }
}

File: code/backend-ms1/src/main/java/ma/zs/univ/service/facade/admin/commun/SecteurAdminService.java
package ma.zs.univ.service.facade.admin.commun;

import java.util.List;
import ma.zs.univ.bean.core.commun.Secteur;
import ma.zs.univ.dao.criteria.core.commun.SecteurCriteria;
import ma.zs.univ.zynerator.service.IService;



public interface SecteurAdminService extends  IService<Secteur,SecteurCriteria>  {

    List<Secteur> findByVilleId(Long id);
    int deleteByVilleId(Long id);
    long countByVilleCode(String code);




}
